src/psyclone/tests/gocean/transformations/globalstoargs_test.py

```diff
@@ -0,0 +1,260 @@
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ----------------------------------------------------------------------------
+# Author: S. Siso, STFC Daresbury Lab
+
+'''Tests the GlobalsToArgumentsTransformation for the  GOcean 1.0 API.'''
+
+from __future__ import absolute_import, print_function
+import os
+import pytest
+from psyclone.parse.algorithm import parse
+from psyclone.psyGen import PSyFactory
+from psyclone.psyir.symbols import DataType
+
+API = "gocean1.0"
+
+BASEPATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                        "..", "..", "test_files")
+
+
+def test_globalstoargumentstrans_wrongapi():
+    ''' Check the KernelGlobalsToArguments with an API other than GOcean1p0'''
+    from psyclone.transformations import KernelGlobalsToArguments, \
+        TransformationError
+
+    trans = KernelGlobalsToArguments()
+    path = os.path.join(BASEPATH, "dynamo0p3")
+    _, invoke_info = parse(os.path.join(path, "1_single_invoke.f90"),
+                           api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3").create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    kernel = invoke.schedule.coded_kernels()[0]
+    with pytest.raises(TransformationError) as err:
+        trans.apply(kernel)
+    assert "The KernelGlobalsToArguments transformation is currently only " \
+           "supported for the GOcean API but got an InvokeSchedule of " \
+           "type:" in str(err.value)
+
+
+def test_globalstoargumentstrans(monkeypatch):
+    ''' Check the KernelGlobalsToArguments transformation '''
+    from psyclone.transformations import KernelGlobalsToArguments, \
+        TransformationError
+    from psyclone.psyGen import Argument
+    from psyclone.psyir.backend.fortran import FortranWriter
+
+    trans = KernelGlobalsToArguments()
+    assert trans.name == "KernelGlobalsToArguments"
+    assert str(trans) == "Convert the global variables used inside the " \
+        "kernel into arguments and modify the InvokeSchedule to pass them" \
+        " in the kernel call."
+
+    # Construct a testing InvokeSchedule
+    _, invoke_info = parse(os.path.join(BASEPATH, "gocean1p0",
+                                        "single_invoke_kern_with_use.f90"),
+                           api=API)
+    psy = PSyFactory(API).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    notkernel = invoke.schedule.children[0]
+    kernel = invoke.schedule.coded_kernels()[0]
+
+    # Monckeypatch resolve_deferred to avoid module searching and importing
+    # in this test. In this case we assume it is a REAL
+    for var in kernel.get_kernel_schedule().symbol_table.global_datasymbols:
```

Comment:
This loop and the similar one at line 170 generate pylint warnings about "cell variable var defined in loop". Please could you check what this means and see whether it can be avoided (in fact, my suggestion above about monkeypatching the class might remove the need for this loop).

Reply:
Fixed by your suggestion above.